credoai/evaluators/ranking_fairness.py

```diff
@@ -1,63 +1,183 @@
-    desired_proportions : dict, Optional
+        A series of the sensitive feature labels (e.g., "male", "female") which should
+        be used to create subgroups
+    rankings : pandas.Series of type int
+        The computed ranks
+        It should be passed to TabularData's y argument with the column name `rankings`
+    scores : pandas.Series of type int or float, Optional
+        A series of the scores
+        It should be passed to TabularData's y argument with the column name `scores`
+    k: int, Optional
+        The top k items are considered as the selected subset
+        If not provided, the top 50% of the items are considered as selected
+    q: float, Optional
+        The relevance score for which items in the pool that have score >= q are "relevant".
+        These two metrics require this to be provided: `qualified_demographic_parity_ratio`
+        and `qualified_balance_ratio`
+    lb_bin: numpy array of shape = (n_bins), Optional
+        The lower bound scores for each bin (bin is greater than or equal to lower bound).
+        These two metrics require this to be provided: `calibrated_demographic_parity_ratio`
+        and `calibrated_balance_ratio`
+    ub_bin: numpy array of shape = (n_bins), Optional
+        The upper bound scores for each bin (bin is less than upper bound).
+        These two metrics require this to be provided: `calibrated_demographic_parity_ratio`
+        and `calibrated_balance_ratio`
+    desired_proportions: dict, Optional
         The desired proportion for each subgroups (e.g., {"male":0.4, "female":0.6})
-        If not provided, equal proportions are used
-    skew_log : bool, Optional
-        If True, will return logarithm of skew values.
+        If not provided, equal proportions are used for calculation of `skew` score
+    down_sampling_step : int, optional
+        down-sampling step for scores empirical distribution curve
+        If not provided, down-sampling is done such that the curve length be nearly 100
     """
 
-    def __init__(self, desired_proportions: dict = None, skew_log: bool = False):
+    def __init__(
+        self,
+        k: int = None,
+        q: float = None,
+        lb_bin: list = None,
+        ub_bin: list = None,
+        desired_proportions: dict = None,
+        down_sampling_step: int = None,
+    ):
         self.desired_proportions = (desired_proportions,)
-        self.skew_log = skew_log
+        self.k = k
+        self.q = q
+        self.down_sampling_step = down_sampling_step
+        if lb_bin is not None and ub_bin is not None:
+            self.lb_bin = np.array(lb_bin)
+            self.ub_bin = np.array(ub_bin)
+        else:
+            self.lb_bin = lb_bin
+            self.ub_bin = ub_bin
 
     required_artifacts = ["data", "sensitive_feature"]
 
     def _setup(self):
-        self.sensitive_features = np.array(self.data.sensitive_feature)
```

Comment:
It's not clear exactly what these changes these are for. They track back to a commit from Amin.

I can't request changes but also can't properly review.